sdk/program/src/vote/state/mod.rs

```diff
@@ -414,13 +427,22 @@ impl VoteState {
             }
         }
 
-        self.epoch_credits.last_mut().unwrap().1 += credits;
+        self.epoch_credits.last_mut().unwrap().1 = self
+            .epoch_credits
+            .last_mut()
+            .unwrap()
+            .1
+            .saturating_add(credits);
     }
 
-    #[allow(clippy::integer_arithmetic)]
     pub fn nth_recent_vote(&self, position: usize) -> Option<&Lockout> {
         if position < self.votes.len() {
-            let pos = self.votes.len() - 1 - position;
+            let pos = self
+                .votes
+                .len()
+                .checked_sub(position)
+                .and_then(|pos| pos.checked_sub(1))
+                .expect("`position` should be less than `self.votes.len()`");
```

Comment:
seem like we can `?` here? if `pos` wraps at runtime, `get()`'s not going to find anything